drivers/starburst/src/metabase/driver/implementation/connectivity.clj

```diff
@@ -121,7 +124,8 @@
                   (update :kerberos-delegation bool->str)
                   (assoc :SSL (:ssl details-map))
                   (assoc :source "Starburst Metabase 4.0.0")
-                  (assoc :clientInfo (if (:impersonation details-map) "impersonate:true" ""))
+                  (assoc-if :clientInfo "impersonate:true" (:impersonation details-map))
+                  (assoc-if :explicitPrepare "false" (:prepared-optimized details-map))
```

Comment:
So we're defaulting to `EXECUTE IMMEDIATE`?

Reply:
If I recall that's Piotr's decision but we can check again with him

Reply:
Turns out the default is NOT `EXECUTE IMMEDIATE` so I've updated a few things in the test. But this part of the code stays the same. `:explicitPrepare "false"` will be added only if the details-map contains `:prepared-optimized true`